### PR TITLE
[new] MST failover

### PR DESCRIPTION
**Pull request by Rajakavitha1**
# Description

Added the failover details from [this article](https://app.intercom.com/a/apps/uvcadzno/articles/articles/2755200/show)

# Links

Fixes #[insert issue link, if any]

# Writing help

For information about style and word usage, see the [style guide](https://docs.timescale.com/timescaledb/latest/contribute-to-docs/)

# Review checklists
Reviewers: use this section to ensure you have checked everything before approving this PR:

## Subject matter expert (SME) review checklist

- [ ] Is the content technically accurate?
- [ ] Is the content complete?
- [ ] Is the content presented in a logical order?
- [ ] Does the content use appropriate names for features and products?
- [ ] Does the content provide relevant links to further information?

## Documentation team review checklist

- [ ] Is the content free from typos?
- [ ] Does the content use plain English?
- [ ] Does the content contain clear sections for concepts, tasks, and references?
- [ ] Have any images been uploaded to the correct location, and are resolvable?
- [ ] If the page index was updated, are redirects required
      and have they been implemented?
- [ ] Have you checked the built version of this content?

### CODE REVIEW

File: mst/failover.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+excerpt: Learn how maintenance is automatically handled on Managed Service for TimescaleDB
+product: mst
+keywords: [maintenance, update, upgrade, failover]
+tags: [failover window]
+---
+
+
+# Failover in Managed Service for TimescaleDB
+
+One standby read-only replica server configured, for Managed Service for
```
Comment (charislam) [inline mst/failover.md:12]:
```suggestion
One standby read-only replica server is configured, for Managed Service for
```

File: mst/failover.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+TimescaleDB on a Pro plan. You can query a read-only replica server, but cannot
+write to a read-only replica server. When a master server fails, standby replica
```
Comment (charislam) [inline mst/failover.md:14]:
```suggestion
write to a read-only replica server. When a master server fails, the standby replica
```

File: mst/failover.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+`replica.servicename.timescaledb.io` works in read-only mode. After the replica
+server promotes itself as master, `servicename.timescaledb.io` points to the new
+master server, and `replica.servicename.timescaledb.io` continues to point to
+the new master server. New replica server is built automatically, and after it
+is in sync, `replica.servicename.timescaledb.io` points to the new replica
+server.
+
+## Controlled failover during upgrades
+
+When applying upgrades or plan changes or business or premium plans, the standby
```
Comment (charislam) [inline mst/failover.md:52]:
```suggestion
When applying upgrades or plan changes on business or premium plans, the standby
```

File: mst/failover.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+server are replaced:
```
Comment (charislam) [inline mst/failover.md:53]:
```suggestion
server is replaced:
```

File: mst/failover.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+A new server is started, and the backup is restored, and the new server starts
```
Comment (charislam) [inline mst/failover.md:55]:
```suggestion
A new server is started, the backup is restored, and the new server starts
```

File: mst/failover.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+following the old master server. After the new server is up and running,
+`replica.servicename.timescaledb.io` is updated, and old replica server is
```
Comment (charislam) [inline mst/failover.md:57]:
```suggestion
`replica.servicename.timescaledb.io` is updated, and the old replica server is
```

File: mst/failover.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+deleted.
+
+For premium plans this step is executed for both replica servers before master
```
Comment (charislam) [inline mst/failover.md:60]:
```suggestion
For premium plans, this step is executed for both replica servers before the master
```

File: mst/failover.md [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+server is replaced. Two new servers are started, backup is restored, and one new
+server is synced up to old master server. When it is time to switch the master
```
Comment (charislam) [inline mst/failover.md:62]:
```suggestion
server is replaced. Two new servers are started, a backup is restored, and one new
server is synced up to the old master server. When it is time to switch the master
```

File: mst/failover.md [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+to a new server, the old master is terminated and one of the new replica servers
+is immediately promoted as a master. At this point `servicename.timescaledb.io`
```
Comment (charislam) [inline mst/failover.md:64]:
```suggestion
is immediately promoted as a master. At this point, `servicename.timescaledb.io`
```

File: mst/failover.md [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+is updated to point at the new master server. Similarly, the new master is
+removed from `replica.servicename.timescaledb.io` record.
```
Comment (charislam) [inline mst/failover.md:66]:
```suggestion
removed from the `replica.servicename.timescaledb.io` record.
```

**Comment by Rajakavitha1** [comment]
Thank you for the edits @charislam !!!!

File: mst/failover.md [hunk 11]
```diff
@@ -0,0 +1,66 @@
+---
+title: Failover
+excerpt: Learn how maintenance is automatically handled on Managed Service for TimescaleDB
+product: mst
+keywords: [maintenance, update, upgrade, failover]
```
Comment (jfjoly) [inline mst/failover.md:5]: I would suggest `HA` and `replica` as additional keywords

**Review by jfjoly — APPROVED** [review]
LGTM. We should consider using `primary` instead of `master`, that's a broader discussion than just this page.

**Comment by Rajakavitha1** [comment]
> LGTM. We should consider using `primary` instead of `master`, that's a broader discussion than just this page.

Yes, this is on our list of things to do.

File: mst/failover.md (hunk 11 above)
Comment (Rajakavitha1) [reply mst/failover.md:5]: done!!!!

File: mst/failover.md [hunk 12]
```diff
@@ -0,0 +1,66 @@
+---
+title: Failover
+excerpt: Learn how maintenance is automatically handled on Managed Service for TimescaleDB
+product: mst
+keywords: [maintenance, update, upgrade, failover, haigh availability, replica]
```
Comment (charislam) [inline mst/failover.md:5]:
```suggestion
keywords: [maintenance, update, upgrade, failover, high availability, replica]
```

**Review by charislam — APPROVED** [review]

**Pull request merged**